src/uu/wc/src/wc.rs

```diff
@@ -688,29 +677,35 @@ fn wc(inputs: &[Input], settings: &Settings) -> UResult<()> {
 
 fn print_stats(
     settings: &Settings,
-    result: &TitledWordCount,
+    result: &WordCount,
+    title: Option<&str>,
     number_width: usize,
 ) -> io::Result<()> {
-    let mut columns = Vec::new();
+    let mut stdout = io::stdout().lock();
 
-    if settings.show_lines {
-        columns.push(format!("{:1$}", result.count.lines, number_width));
-    }
-    if settings.show_words {
-        columns.push(format!("{:1$}", result.count.words, number_width));
-    }
-    if settings.show_chars {
-        columns.push(format!("{:1$}", result.count.chars, number_width));
-    }
-    if settings.show_bytes {
-        columns.push(format!("{:1$}", result.count.bytes, number_width));
-    }
-    if settings.show_max_line_length {
-        columns.push(format!("{:1$}", result.count.max_line_length, number_width));
-    }
-    if let Some(title) = &result.title {
-        columns.push(title.clone());
+    let mut space = "";
+
+    macro_rules! print_column {
```

Comment:
Here again the macro is not really pulling its weight in my opinion. It could also be expressed as a closure:
```rust
let mut space = "";

let mut print_column = |show: bool, result: usize| -> io::Result<()> {
    if show {
        write!(stdout, "{space}{:width$}", result, width = number_width)?;
        space = " ";
    }
    Ok(())
};

print_column(settings.show_lines, result.lines)?;
print_column(settings.show_words, result.words)?;
print_column(settings.show_chars, result.chars)?;
print_column(settings.show_bytes, result.bytes)?;
print_column(settings.show_max_line_length, result.max_line_length)?;
```